### ISSUE TITLE
Cookies file does not get created by get-maps-cookies

### ISSUE THREAD

**Issue by jezzaaa** [costastf/locationsharinglib#29]
I have v2.0.11 installed, and an account with 2FA. When I run get-maps-cookies, it doesn't create a cookies file, whether I specify the cookie filename or leave it to choose the default. I know that it authenticates me correctly because a) if bombs out if I use the wrong password, and b) my phone prompts me to confirm that it was me.

I've tried letting the sample script create the cookies file, which it does, but then the next time I run the script, it doesn't like the cookies file.

**Comment by jezzaaa** [comment]
Oh yeah, I know it's not creating the cookies file in some weird location because, a) I used "find . -cmin -10 -ls" to list all new files in my home dir, and b) I ran it through strace and grepped for "open" and the last file opened was cacert.pem for reading.

**Comment by costastf** [comment]
In order for me to help i would need some more info. Could you please paste a console output of everything that you do retracting your credentials of course? In the latest version i have added logging capabilities on the cli tool, please execute the tool with debug and post result.

**Comment by jezzaaa** [comment]
Hmm, I just did an upgrade to v3.0.2 and it no longer runs at all :-( :

````
$ workon locationsharinglib
(locationsharinglib) $ pip list | grep locationsharinglib
locationsharinglib 3.0.2     /home/jpl/.virtualenvs/locationsharinglib/lib/python2.7/site-packages
(locationsharinglib) $ get-maps-cookies --email email@example.com --password example
Traceback (most recent call last):
  File "/home/jpl/.virtualenvs/locationsharinglib/bin/get-maps-cookies", line 11, in <module>
    sys.exit(main())
  File "/home/jpl/.virtualenvs/locationsharinglib/local/lib/python2.7/site-packages/locationsharinglib/cli.py", line 136, in main
    CookieGetter(args.email, args.password, args.cookies_file)
  File "/home/jpl/.virtualenvs/locationsharinglib/local/lib/python2.7/site-packages/locationsharinglib/locationsharinglib.py", line 327, in __init__
    super(CookieGetter, self).__init__(email,
TypeError: super() argument 1 must be type, not classobj
(locationsharinglib) $ python --version
Python 2.7.12
(locationsharinglib) $ type python
python is hashed (/home/jpl/.virtualenvs/locationsharinglib/bin/python)
````
I don't know enough python to debug this.

**Comment by costastf** [comment]
In version 3 i have dropped support for python2 as this is mostly made with HA in mind and that is python3 and its about time we move forward. That is why the major bump on the code. If you use python 3 it will work for you just fine.

**Comment by jezzaaa** [comment]
Ah, OK, thankyou!

Can I suggest the install documentation use "mkvirtualenv -p python3 ..."?

Now I have get-maps-cookies running and making a cookies file. To use the cookies file, do I just call Service() with empty username and password? I want to avoid having a hard-coded password in a script.

Also, my test script gets this error:

```
(locationsharinglib) $ python3 ./test.py
Traceback (most recent call last):
  File "./test.py", line 7, in <module>
    for person in service.get_all_people():
  File "/home/jpl/.virtualenvs/locationsharinglib/lib/python3.5/site-packages/locationsharinglib/locationsharinglib.py", line 459, in get_all_people
    people = self.get_shared_people() + [self.get_authenticated_person()]
  File "/home/jpl/.virtualenvs/locationsharinglib/lib/python3.5/site-packages/locationsharinglib/locationsharinglib.py", line 421, in get_shared_people
    for info in output[0]:
TypeError: 'NoneType' object is not iterable
```

Does this simply mean that nobody is sharing their location with this account? This is not the case.

**Comment by jezzaaa** [comment]
OK, seem to have gone backwards. If I try to re-create the cookie file, I get no message on my phone asking me to confirm the login like before. Even when I intentionally use the wrong password, the cookie file gets created with no error. In fact, this appears to be the exact same behaviour if I put in a bogus email address (bogus@example.com).

**Comment by jezzaaa** [comment]
And now I'm only getting the "locationsharinglib.locationsharinglibexceptions.NoExpectedFormOption" message. I tried adding "-L DEBUG" but nothing shows. I haven't changed anything, so I'm perplexed why it's changed behaviour.

**Comment by jezzaaa** [comment]
[excuse my low standard of python knowledge]

OK, so using "print" (and cleaning up utf-8 chars) I dumped the response just before the NoExpectedFormOption exception is raised. The content seems OK to me. I see the following:

```
<form novalidate method="post
" action="https://accounts.google.com/signin/challenge/sl/password" id="gaia_loginform">\n  <input name="Page" type="hidden" value="Pass
wordSeparationSignIn">
```
The code in _get_required_form() seems to be looking for `<form` and then `action=` so it all seems to be correct. I can't work out why the exception is being raised. Not sure what I can do next to investigate.

Edit: I think I get it. The "action" attribute exist, but it also needs to evaluate to "Tap Yes", or something like that. I'm guessing that form.get('action') gets the value of the "action=..." URI?

**Comment by costastf** [comment]
What that means is that your account is not setup for a push challenge but it only accepts a password challenge. That is not supported as I do not think that making the library interactive makes so much sense.
The reason why you did not get any debug output is because I am essentially dumb and did not configure the logging correctly. That is fixed on v3.0.5. If you could give that a go with debug logging we should have much more output to work with.

**Comment by costastf** [comment]
Also any change on the documentation that you would think is helpful is greatly appreciated and I will accepts MRs.

**Comment by jezzaaa** [comment]
> What that means is that your account is not setup for a push challenge but it only accepts a password challenge. That is not supported as I do not think that making the library interactive makes so much sense.

Agreed that it needs to be non-interactive. But I thought once I had responded positively to the push, the cookie would work non-interactively from then on. I guess I should try changing my authentication method.

> The reason why you did not get any debug output is because I am essentially dumb and did not configure the logging correctly.

LOL 👍 :-)

> That is fixed on v3.0.5. If you could give that a go with debug logging we should have much more output to work with.

Awesome, will do, thanks.

**Comment by jezzaaa** [comment]
Debug still not working for me. Am I using it correctly?
```
(locationsharinglib) $ pip list | grep locationsharinglib
locationsharinglib 3.0.5
(locationsharinglib) $ get-maps-cookies --email email@example.com --password example -L DEBUG
(locationsharinglib) $
```

**Comment by jezzaaa** [comment]
> Also any change on the documentation that you would think is helpful is greatly appreciated and I will accepts MRs.

Sorry, can you define "MR"? Google isn't helping at all.

**Comment by costastf** [comment]
Merge requests. That is a gitlab term. Github calls it PR. I will look into the debug issue today hopefully and report back here.

**Comment by costastf** [comment]
Please try 3.0.6 where I added the response to the exception raised. Be careful to removed personal info before posting here.

**Comment by jezzaaa** [comment]
> Please try 3.0.6 where I added the response to the exception raised. Be careful to removed personal info before posting here.

Upgraded to 3.0.6, but still no output from get-maps-cookies at all, even with "-L DEBUG". I get a cookie file, but running test.py still gives "TypeError: 'NoneType' object is not iterable".

**Comment by costastf** [comment]
Sorry, what is test.py?

**Comment by costastf** [comment]
This should be fixed with 3.0.11 which renders cookie getter obsolete.
